dbfopen.c

```diff
@@ -68,6 +68,24 @@ CPL_INLINE static void CPL_IGNORE_RET_VAL_INT(CPL_UNUSED int unused)
 #define CPL_IGNORE_RET_VAL_INT(x) x
 #endif
 
+/************************************************************************/
+/*                           DBFHandleFree()                            */
+/*                                                                      */
+/*      This is called to free out various pointers inside              */
+/*      DBFHandle structure.                                            */
+/************************************************************************/
+static void DBFHandleFree(DBFHandle psDBF)
+{
+    free(psDBF->panFieldOffset);
+    free(psDBF->panFieldSize);
+    free(psDBF->panFieldDecimals);
+    free(psDBF->pachFieldType);
+    free(psDBF->pszHeader);
+    free(psDBF->pszCurrentRecord);
+    free(psDBF->pszWorkField);
+    free(psDBF->pszCodePage);
+}
+
```

Comment:
I don't think this is needed. And as it, given that it doesn't nullify the fields, it would mostly certainly lead to double free:
```suggestion
```

Reply:
I agree with your point that we are not nullifying the fields. I can add this part.

But, I am a bit confused about why we don't need this function?

Let's take `DBFAddNativeFieldType` function for example and we have filled few fields in the `psDBF` function before realloc fails. Now, is the onus on the code which calls this function and sees that it has returned due to error and free out `psDBF` handle? If not, shouldn't we take care of it in the code here?

Reply:
> But, I am a bit confused about why we don't need this function?

My thought is that in the situation where we would need it, we are pretty much busted and we would leave the library in a inconsistent state. I believe that in the places where we can't safely reset the library state to a consistent state because an intermediate memory allocation failed, we would be much better assert()'ing. Memory allocation failures in dbfopen are really unlikely as the data structure size is really small.

> Now, is the onus on the code which calls this function and sees that it has returned due to error and free out psDBF handle? If not, shouldn't we take care of it in the code here?

Yes the onus is on the caller to check the return code and decide what to do with it. The  DBFAddNativeFieldType() function shouldn't free psDBF itself. It should leave it though in a consistent state